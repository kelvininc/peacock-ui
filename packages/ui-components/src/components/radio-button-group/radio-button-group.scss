@import '../../assets/styles/globals';

:host > kv-radio-button {
	margin-right: $spacing;
}

:host > kv-radio-button:last-child {
	margin-right: unset;
}
