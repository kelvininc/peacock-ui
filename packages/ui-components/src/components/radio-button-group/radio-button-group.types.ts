import { EventEmitter } from '@stencil/core';

export interface IRadioButtonGroupButton {
	/** (required) Adds a label aside the button */
	label: string;
	/** (optional) The value to be emitted upon click events */
	value?: string;
	/** (optional) Sets the button's styling to be disabled and disables click events */
	disabled?: boolean;
}

export interface IRadioButtonGroup {
	/** (optional) List of radio buttons */
	buttons: IRadioButtonGroupButton[];
	/** (optional) A record with the button's label/value and with if the selected value */
	selectedButtons?: Record<string, boolean>;
	/** (optional) Disables all buttons */
	disabled?: boolean;
}

export interface IRadioButtonGroupEvents {
	/** When the radio button selection changes, emit the requested tab's key */
	checkedChange: EventEmitter<string>;
}
