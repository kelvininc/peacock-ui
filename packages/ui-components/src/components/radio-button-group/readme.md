# _<kv-radio-button-group>_

<!-- Auto Generated Below -->


## Properties

| Property          | Attribute  | Description                                                                      | Type                        | Default |
| ----------------- | ---------- | -------------------------------------------------------------------------------- | --------------------------- | ------- |
| `buttons`         | --         | (optional) List of radio buttons                                                 | `IRadioButtonGroupButton[]` | `[]`    |
| `disabled`        | `disabled` | (optional) Disables all buttons                                                  | `boolean`                   | `false` |
| `selectedButtons` | --         | (optional) A record with the button's label/value and with if the selected value | `{ [x: string]: boolean; }` | `{}`    |


## Events

| Event           | Description                                                           | Type                  |
| --------------- | --------------------------------------------------------------------- | --------------------- |
| `checkedChange` | When the radio button selection changes, emit the requested tab's key | `CustomEvent<string>` |


## Dependencies

### Depends on

- [kv-radio-button](../radio-button)

### Graph
```mermaid
graph TD;
  kv-radio-button-group --> kv-radio-button
  style kv-radio-button-group fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


