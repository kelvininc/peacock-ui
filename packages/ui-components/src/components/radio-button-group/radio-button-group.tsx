import { Component, h, Prop, Host, Event, EventEmitter } from '@stencil/core';
import { IRadioButton } from '../radio-button/radio-button.types';
import { IRadioButtonGroup, IRadioButtonGroupEvents } from './radio-button-group.types';

@Component({
	tag: 'kv-radio-button-group',
	styleUrl: 'radio-button-group.scss',
	shadow: true
})
export class KvRadioButtonGroup implements IRadioButtonGroup, IRadioButtonGroupEvents {
	/** @inheritdoc */
	@Prop({ reflect: true }) buttons: IRadioButton[] = [];
	/** @inheritdoc */
	@Prop({ reflect: true }) selectedButtons?: Record<string, boolean> = {};
	/** @inheritdoc */
	@Prop({ reflect: true }) disabled?: boolean = false;

	/** @inheritdoc */
	@Event() checkedChange: EventEmitter<string>;

	render() {
		return (
			<Host>
				{this.buttons.map(button => (
					<kv-radio-button
						exportparts="radio-button"
						value={button.value}
						label={button.label}
						icon={button.icon}
						download={button.download}
						href={button.href}
						target={button.target}
						preventDefault={button.preventDefault}
						disabled={this.disabled || button.disabled}
						checked={this.selectedButtons[button.value] === true}
					></kv-radio-button>
				))}
			</Host>
		);
	}
}
