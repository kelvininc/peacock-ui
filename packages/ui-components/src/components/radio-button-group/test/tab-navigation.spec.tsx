import { newSpecPage, SpecPage } from '@stencil/core/testing';
import { cloneDeep } from 'lodash-es';
import { KvRadioButtonGroup } from '../radio-button-group';
import { RADIO_BUTTON_ITENS } from './radio-button-group.mock';
import { h } from '@stencil/core';

describe('Radio Button Group (unit tests)', () => {
	let page: SpecPage;
	let component: KvRadioButtonGroup;

	const radioButtonsMock = cloneDeep(RADIO_BUTTON_ITENS);

	describe('when rendering with required props', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvRadioButtonGroup],
				template: () => <kv-radio-button-group buttons={radioButtonsMock}></kv-radio-button-group>
			});
			component = page.rootInstance;
		});

		it('should match the snapshot', () => {
			expect(page.root).toMatchSnapshot();
		});

		it('should set the correct component tabs', () => {
			expect(component.buttons).toEqual([
				{
					label: 'Option 1',
					id: 'opt1'
				},
				{
					label: 'Option 2',
					id: 'opt2'
				},
				{
					label: 'Option 3',
					id: 'opt3',
					disabled: true
				},
				{
					label: 'Option 4',
					id: 'opt4',
					active: true
				}
			]);
		});
	});
});
