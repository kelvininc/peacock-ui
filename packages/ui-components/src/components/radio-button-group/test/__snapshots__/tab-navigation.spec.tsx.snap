// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Button Group (unit tests) when rendering with required props should match the snapshot 1`] = `
<kv-radio-button-group>
  <mock:shadow-root>
    <kv-radio-button id="0" label="Option 1"></kv-radio-button>
    <kv-radio-button id="1" label="Option 2"></kv-radio-button>
    <kv-radio-button disabled="" id="2" label="Option 3"></kv-radio-button>
    <kv-radio-button checked="" id="3" label="Option 4"></kv-radio-button>
  </mock:shadow-root>
</kv-radio-button-group>
`;
